reconcile/external_resources/model.py

```diff
@@ -60,30 +65,48 @@ def state_path(self) -> str:
         return f"{self.provision_provider}/{self.provisioner_name}/{self.provider}/{self.identifier}"
 
 
+SUPPORTED_RESOURCE_PROVIDERS = NamespaceTerraformProviderResourceAWSV1
+SUPPORTED_RESOURCE_TYPES = NamespaceTerraformResourceRDSV1
+
+
 class ExternalResourcesInventory(MutableMapping):
     _inventory: dict[ExternalResourceKey, ExternalResourceSpec] = {}
 
+    def _build_external_resource_spec(
+        self,
+        namespace: NamespaceV1,
+        provider: SUPPORTED_RESOURCE_PROVIDERS,
+        resource: SUPPORTED_RESOURCE_TYPES,
+    ) -> ExternalResourceSpec:
+        spec = ExternalResourceSpec(
+            provision_provider=provider.provider,
+            provisioner=provider.provisioner.dict(),
+            resource=resource.dict(
+                exclude={
+                    FLAG_RESOURCE_MANAGED_BY_ERV2,
+                    FLAG_DELETE_RESOURCE,
+                    MODULE_OVERRIDES,
+                }
+            ),
+            namespace=namespace.dict(),
+        )
+        spec.metadata[FLAG_DELETE_RESOURCE] = resource.delete
+        spec.metadata[MODULE_OVERRIDES] = resource.module_overrides
```

Comment:
can `metadata` be set in constructor directly like?

```
ExternalResourceSpec(
    ...
    metadata={
        FLAG_DELETE_RESOURCE: resource.delete,
        MODULE_OVERRIDES: module_overrides
    }
)
```

Reply:
Nope, metadata is not part of the constructor. It's excluded because `ExternalResourceSpec` is shared with `terraform-resources`.